sequence_processing_pipeline/Pipeline.py

```diff
@@ -100,6 +99,15 @@ def __init__(self, configuration_file_path, run_id, output_path,
             raise PipelineError(
                 'older_than cannot be equal to or less than younger_than.')
 
+        # required files for successful operation
+        # both RTAComplete.txt and RunInfo.xml should reside in the root of
+        # the run directory.
+        required_files = ['RTAComplete.txt', 'RunInfo.xml']
+        for some_file in required_files:
+            if not exists(join(self.run_dir, some_file)):
```

Comment:
Should we also check that is readable or just existence should do the trick?

Reply:
The files I encountered during development were just zero-length files so nothing much to check; existence should be fine.

Reply:
Note that size of a file is independent of the permissions; in other words a zero-length file can still be or not be readable.

Reply:
Confirmed via testing that os.path.exists() doesn't need to be able to read the file to return True for a file that exists. I imagine as long as the execute bit is set on the parent directory, exists() should be able to work as intended.

Reply:
Just to clarify, RTAComplete.txt is just a zero-length file that acts a sentinel. Its presence means that the directory is ready to be processed. If the file is not present at this point, it's going to fail saying that required file 'RTAComplete.txt' is not present, which is accurate. There's never going to be a time where we want to read this file.

RunInfo.xml is a little different in that the file needs to be readable and be a valid RunInfo.xml file. There is the possibility that a bad RunInfo.xml or an unreadable RunInfo.xml passes this test and later causes bcl2fastq/bcl-convert to choke. Perhaps the prudent thing would be to add a check to ensure RunInfo.xml is readable, but leave the validation of it to bcl2fastq.